### PR TITLE
Log levels

### PR DESCRIPTION
**Pull request by phunkyfish**
## Description

With the new SPD logger NOTICE and SERVERE map to INFO and FATAL respectively and they can be removed.

Valid set of log levels should now be: `DEBUG, INFO, WARNING, ERROR and FATAL`

## Motivation and Context

No longer need so many log levels. End state would be to remove them completely but python/binary addons would all need to be updated first.

## How Has This Been Tested?

OSX 

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [X] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Review by enen92 — COMMENTED** [review]
Not commenting the actual change, just the docs for python:

`ModuleXbmc.h` needs to remove the `xbmc.LOGNOTICE` (LOGSEVERE was never there?) from the table and add a revision tag at the end. I also improved a bit the docs, below the complete block for quick copy-pasta:

```
#ifdef DOXYGEN_SHOULD_USE_THIS
    ///
    /// \ingroup python_xbmc
    /// @brief \python_func{ xbmc.log(msg[, level]) }
    ///-----------------------------------------------------------------------
    /// Write a string to Kodi's log file and the debug window.
    ///
    /// @param msg                 string - text to output.
    /// @param level               [opt] integer - log level to output at.
    ///                            <em>(default=LOGDEBUG)</em>
    ///  |  Value:         | Description:                                      |
    ///  |----------------:|---------------------------------------------------|
    ///  | xbmc.LOGDEBUG   | In depth information about the status of Kodi. This information can pretty much only be deciphered by a developer or long time Kodi power user.
    ///  | xbmc.LOGINFO    | Something has happened. It's not a problem, we just thought you might want to know. Fairly excessive output that most people won't care about.
    ///  | xbmc.LOGWARNING | Something potentially bad has happened. If Kodi did something you didn't expect, this is probably why. Watch for errors to follow.
    ///  | xbmc.LOGERROR   | This event is bad. Something has failed. You likely noticed problems with the application be it skin artifacts, failure of playback a crash, etc.
    ///  | xbmc.LOGFATAL   | We're screwed. Kodi is about to crash.
    ///
    /// @note Addon developers are advised to keep LOGDEBUG as the default logging level and to use conservative logging
    ///       (log only if needed). Excessive logging makes it harder to debug kodi itself.
    ///
    /// Logging in kodi has a global configuration level that controls how text is written to the log. This global logging
    ///         behaviour can be changed in the GUI (Settings -> System -> Logging) (debug toggle) or furthered configured
    ///         in advancedsettings (loglevel setting). Text is written to the log for the following conditions:
    ///
    ///           - loglevel == -1 (NONE, nothing at all is logged to the log)
    ///           - loglevel == 0 (NORMAL, shows LOGINFO, LOGERROR and LOGFATAL) - Default kodi behaviour
    ///           - loglevel == 1 (DEBUG, shows all) - Behaviour if you toggle debug log in the GUI
    ///
    ///
    /// ------------------------------------------------------------------------
    /// @python_v17 Default level changed from LOGNOTICE to LOGDEBUG
    /// @python_v19 Removed LOGNOTICE (use LOGINFO) and LOGSEVERE (use LOGFATAL)
    ///
    /// **Example:**
    /// ~~~~~~~~~~~~~{.py}
    /// ..
    /// xbmc.log(msg='This is a test string.', level=xbmc.LOGDEBUG);
    /// ..
    /// ~~~~~~~~~~~~~
    ///
    log(...);
```

**Comment by phunkyfish** [comment]
Thanks @enen92

**Comment by da-anda** [comment]
> below the complete block for quick copy-pasta

my favorite pasta 🤤

File: xbmc/interfaces/legacy/ModuleXbmc.h [hunk 1]
```diff
@@ -52,25 +52,25 @@ namespace XBMCAddon
...
+    ///           - loglevel == 0 (NORMAL, shows LOGINFO, LOGWARNING, LOGERROR and LOGFATAL) - Default kodi behaviour
+    ///           - loglevel == 1 (DEBUG, shows all) - Behaviour if you toggle debug log in the GUI
     ///
     ///
     /// ------------------------------------------------------------------------
     /// @python_v17 Default level changed from LOGNOTICE to LOGDEBUG
+    /// @python_v19 Removed LOGNOTICE (use LOGINFO) and LOGSEVERE (use LOGFATAL)
     ///
     /// **Example:**
     /// ~~~~~~~~~~~~~{.py}
```
Comment (AlwinEsch) [inline xbmc/interfaces/legacy/ModuleXbmc.h:76]: About the docs are not so much spaces needed, also can the line length a bit smaller, here a example how I thought:
```cpp
    ///
    /// \ingroup python_xbmc
    /// @brief \python_func{ xbmc.log(msg[, level]) }
    ///-----------------------------------------------------------------------
    /// Write a string to Kodi's log file and the debug window.
    ///
    /// @param msg                 string - text to output.
    /// @param level               [opt] integer - log level to output at.
    ///                            <em>(default=LOGDEBUG)</em>
    ///  |  Value:         | Description:                                      |
    ///  |----------------:|---------------------------------------------------|
    ///  | xbmc.LOGDEBUG   | In depth information about the status of Kodi. This information can pretty much only be deciphered by a developer or long time Kodi power user.
    ///  | xbmc.LOGINFO    | Something has happened. It's not a problem, we just thought you might want to know. Fairly excessive output that most people won't care about.
    ///  | xbmc.LOGWARNING | Something potentially bad has happened. If Kodi did something you didn't expect, this is probably why. Watch for errors to follow.
    ///  | xbmc.LOGERROR   | This event is bad. Something has failed. You likely noticed problems with the application be it skin artifacts, failure of playback a crash, etc.
    ///  | xbmc.LOGFATAL   | We're screwed. Kodi is about to crash.
    ///
    /// @note Addon developers are advised to keep `LOGDEBUG` as the default
    /// logging level and to use conservative logging (log only if needed).
    /// Excessive logging makes it harder to debug kodi itself.
    ///
    /// Logging in kodi has a global configuration level that controls how text
    /// is written to the log. This global logging behaviour can be changed in
    /// the GUI (**Settings -> System -> Logging**) (debug toggle) or furthered
    /// configured in advancedsettings (loglevel setting).
    ///
    /// Text is written to the log for the following conditions:
    ///  - loglevel == -1 (NONE, nothing at all is logged to the log)
    ///  - loglevel == 0 (NORMAL, shows `LOGINFO`, `LOGWARNING`, `LOGERROR` and `LOGFATAL`) - Default kodi behaviour
    ///  - loglevel == 1 (DEBUG, shows all) - Behaviour if you toggle debug log in the GUI
    ///
    ///
    /// ------------------------------------------------------------------------
    /// @python_v17 Default level changed from `LOGNOTICE` to `LOGDEBUG`
    /// @python_v19 Removed `LOGNOTICE` (use `LOGINFO`) and `LOGSEVERE` (use `LOGFATAL`)
    ///
    /// **Example:**
    /// ~~~~~~~~~~~~~{.py}
    /// ..
    /// xbmc.log(msg='This is a test string.', level=xbmc.LOGDEBUG);
    /// ..
    /// ~~~~~~~~~~~~~
    ///
    log(...);
```

Here his generated doxygen docs:
Bildschirmfoto vom 2020-04-23 01-24-53

**Review by DaveTBlake — APPROVED** [review]

File: xbmc/interfaces/legacy/ModuleXbmc.h (hunk 1 above)
Comment (phunkyfish) [reply xbmc/interfaces/legacy/ModuleXbmc.h:79]: Cool, changed @AlwinEsch!

**Review by AlwinEsch — APPROVED** [review]

**Comment by jakkerson** [comment]
xbmc.LOGINFO does not work with kodi 18.7 tho?

In this snippet ferm 18.7

`        xbmc.log("TEST THIS NOTICE", xbmc.LOGNOTICE)
        xbmc.log("TEST THIS INFO", xbmc.LOGINFO)
        xbmc.log("TEST THIS ERROR", xbmc.LOGERROR)`

We only get this back

`2020-06-13 12:57:29.720 T:7288  NOTICE: UDP: Listening on port 9777 (ipv6 : false)
2020-06-13 12:57:30.587 T:896  NOTICE: TEST THIS NOTICE
2020-06-13 12:57:30.587 T:896   ERROR: TEST THIS ERROR
2020-06-13 12:57:31.822 T:9820   ERROR: CImageLoader::DoWork - Direct texture file loading failed for `

So doesnt this make it a little counter productive for developers who are trying to maintain PY2/3 compatibility on single addon bases?

**Pull request merged**